File: test/end-to-end/Graphs/C#/Data/Edge.cs
namespace Graphs.Data
{
    public class Edge<T>
    {
        public double Distance;
        public WeightedNode<T> From;
        public WeightedNode<T> To;

        public Edge(double distance, WeightedNode<T> from, WeightedNode<T> to)
        {
            this.Distance = distance;
            this.From = from;
            this.To = to;
        }
    }
}

File: test/end-to-end/Graphs/Java/Data/Edge.java
package graphs.data;

public class Edge<T> {
    public Double distance;
    public WeightedNode<T> from;
    public WeightedNode<T> to;

    public Edge(Double distance, WeightedNode<T> from, WeightedNode<T> to) {
        this.distance = distance;
        this.from = from;
        this.to = to;
    }
}
